vortex-tui/src/browse/mod.rs

```diff
@@ -98,13 +165,38 @@ fn handle_normal_mode(app: &mut AppState, event: Event) -> HandleResult {
                 }
                 (KeyCode::Left | KeyCode::Char('h'), _)
                 | (KeyCode::Char('b'), KeyModifiers::CONTROL) => {
-                    if app.current_tab == Tab::Layout {
-                        // Ascend back up to the Parent node
-                        app.cursor = app.cursor.parent();
-                        // Reset the list scroll state.
-                        app.layouts_list_state = ListState::default().with_selected(Some(0));
+                    match app.current_tab {
+                        Tab::Layout => {
+                            // Ascend back up to the Parent node
+                            app.cursor = app.cursor.parent();
+                            // Reset the list scroll state.
+                            app.layouts_list_state = ListState::default().with_selected(Some(0));
+                        }
+                        Tab::Segments => {
+                            app.segment_grid_state.horizontal_scroll =
+                                app.segment_grid_state.horizontal_scroll.saturating_sub(20);
+                            app.segment_grid_state.horizontal_scroll_state = app
+                                .segment_grid_state
+                                .horizontal_scroll_state
+                                .position(app.segment_grid_state.horizontal_scroll);
+                        }
                     }
                 }
+                (KeyCode::Right | KeyCode::Char('l'), _)
+                | (KeyCode::Char('b'), KeyModifiers::ALT) => match app.current_tab {
+                    Tab::Layout => {}
+                    Tab::Segments => {
+                        app.segment_grid_state.horizontal_scroll = app
+                            .segment_grid_state
+                            .horizontal_scroll
+                            .saturating_add(20)
+                            .min(app.segment_grid_state.max_horizontal_scroll);
+                        app.segment_grid_state.horizontal_scroll_state = app
+                            .segment_grid_state
+                            .horizontal_scroll_state
+                            .position(app.segment_grid_state.horizontal_scroll);
```

Comment:
this should be abstracted with a function. Annoyingly in ratatui ListScroll has the useful methods while ScrollbarState only has previous and next